drivers/net/mvpp2/mrvl_ptp.c

```diff
@@ -0,0 +1,433 @@
+#include "mrvl_ptp.h"
+
+#include <fcntl.h>
+#include <linux/ethtool.h>
+#include <linux/sockios.h>
+#include <net/if.h>
+#include <net/if_arp.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include <rte_cycles.h>
+#include <rte_alarm.h>
+
+#define CLOCKFD 3
+#define FD_TO_CLOCKID(fd)	((clockid_t) ((((unsigned int) ~fd) << 3) | CLOCKFD))
+#define CLOCKID_TO_FD(clk)	((unsigned int) ~((clk) >> 3))
+
+/* enable timestamp in mbuf */
+static bool mrvl_enable_ts[RTE_MAX_ETHPORTS];
+static uint64_t mrvl_timestamp_rx_dynflag;
+static int mrvl_timestamp_dynfield_offset = -1;
+
+static inline rte_mbuf_timestamp_t *
```

Comment:
you probably want to move it (and many other functions - if not all) to the header.

Reply:
I didn't want, but it seems I will have to